build-system/tasks/dist.js

```diff
@@ -55,6 +60,37 @@ const {maybeUpdatePackages} = require('./update-packages');
 const {green, cyan} = colors;
 const argv = require('minimist')(process.argv.slice(2));
 
+const babel = require('@babel/core');
+const deglob = require('globs-to-files');
+
+function transferSrcsToTempDir() {
+  const tmp = `${tempy.directory()}/amphtml`;
+  process.env.AMP_TMP_DIR = tmp;
+  log(green(`creating temp directory: ${tmp}`));
+
+  if (!isTravisBuild()) {
+    log('Writing transforms to', colors.cyan(tmp));
+  }
+  const files = deglob.sync(BABEL_SRC_GLOBS);
+  log(green('Executing babel transforms'));
+  files.forEach(file => {
+    if (file.startsWith('node_modules/') || file.startsWith('third_party/')) {
+      fs.copySync(file, `${tmp}/${file}`);
+      return;
+    }
+
+    const {code} = babel.transformFileSync(file, {
+      plugins: conf.plugins({
```

Comment:
Use the `compact: false` setting to avoid [this](https://travis-ci.org/ampproject/amphtml/jobs/556410703#L410) noisy warning:
```
[BABEL] Note: The code generator has deoptimised the styling of /home/travis/build/ampproject/amphtml/third_party/react-dates/bundle.js as it exceeds the max of 500KB.
```